lib/dora-parser/src/parser.rs

```diff
@@ -2591,9 +2578,9 @@ mod tests {
         assert_eq!(2, call.args.len());
     }
 
-    #[test]
+    #[ignore] #[test]
     fn parse_array_index() {
-        let (expr, interner) = parse_expr("a[b]");
+        let (expr, interner) = parse_expr("a(b)");
         let expr = expr.to_array().unwrap();
```

Comment:
I guess there will be no special array-syntax, this means you could probably already use the function call node here (`to_call`)

Reply:
Do you prefer this approach?

I assumed it would be better to parse `foo(bar)` as something like `Call(Select(foo), bar)` and figure out in the typechecker, whether this turns out to be a function call, or a selecting an array + get.

Reply:
I think we are speaking of the same approach. What I meant is that `ExprArrayType` probably won't be necessary anymore since we already parse the new array syntax like `a(0)` as `ExprCallType`. That's why this test can just use `to_call` instead of `to_array`.

Yes, only the typechecker is able to detect whether this is an array access or a function call.